meos/src/general/lifting.c

```diff
@@ -1551,211 +1543,438 @@ tfunc_temporal_temporal(const Temporal *temp1, const Temporal *temp2,
   if (! over_span_span(&s1, &s2))
     return NULL;
 
-  Temporal *result = NULL;
   assert(temptype_subtype(temp1->subtype));
   assert(temptype_subtype(temp2->subtype));
-  if (temp1->subtype == TINSTANT)
+  switch (temp1->subtype)
   {
-    if (temp2->subtype == TINSTANT)
-      result = (Temporal *) tfunc_tinstant_tinstant(
-        (TInstant *) temp1, (TInstant *) temp2, lfinfo);
-    else if (temp2->subtype == TSEQUENCE)
-      result = MEOS_FLAGS_DISCRETE_INTERP(temp2->flags) ?
-        (Temporal *) tfunc_tinstant_tdiscseq(
-          (TInstant *) temp1, (TSequence *) temp2, lfinfo) :
-        (Temporal *) tfunc_tinstant_tcontseq(
-          (TInstant *) temp1, (TSequence *) temp2, lfinfo);
-    else /* temp2->subtype == TSEQUENCESET */
-      result = (Temporal *) tfunc_tinstant_tsequenceset(
-        (TInstant *) temp1, (TSequenceSet *) temp2, lfinfo);
+    case TINSTANT:
+    {
+      switch (temp2->subtype)
+      {
+        case TINSTANT:
+          return (Temporal *) tfunc_tinstant_tinstant(
+            (TInstant *) temp1, (TInstant *) temp2, lfinfo);
+        case TSEQUENCE:
+          return MEOS_FLAGS_DISCRETE_INTERP(temp2->flags) ?
+            (Temporal *) tfunc_tinstant_tdiscseq(
+              (TInstant *) temp1, (TSequence *) temp2, lfinfo) :
+            (Temporal *) tfunc_tinstant_tcontseq(
+              (TInstant *) temp1, (TSequence *) temp2, lfinfo);
+        case TSEQUENCESET:
+          return (Temporal *) tfunc_tinstant_tsequenceset(
+            (TInstant *) temp1, (TSequenceSet *) temp2, lfinfo);
+        default:
+          meos_error(ERROR, MEOS_ERR_INTERNAL_ERROR,
+            "Unexpected temporal subtype: %d", temp2->subtype);
+          return NULL;
```

Comment:
We do not need the default in the switch case.
Meos errors should be used for runtime errors, such as wrong parameters, invalid sequence lengths and similar stuff..
The default check here is to test for correctness, which should only be done with the asserts (already present above) in debug mode. At runtime, this default will never be hit, unless there is a bug, which we can detect in debug mode.

This is also true for all other switch cases on subtype.

Reply:
Solved